framework/src/geomsearch/PenetrationLocator.C

```diff
@@ -127,7 +127,13 @@ PenetrationLocator::detectPenetration()
 void
 PenetrationLocator::reinit()
 {
+  // Delete the PenetrationInfo objects we own before clearing the
+  // map, or we have a memory leak.
+  for (auto & it : _penetration_info)
+    delete it.second;
```

Comment:
"with limited scope", you say?  I see that, like me, you saw a leading underscore in the name of a concrete data structure and you assumed that this variable had been defined as private by a professional software developer rather than as public by a madman.  I regret to say that our assumption is incorrect.

The _penetration_info map is public, is used directly by one or two dozen files, and those uses include at least 13 places in modules/ where the user code assumes that the map values can be directly assigned to ```PenetrationInfo*``` without any silly ```.get()``` call involved.  I assumed there would be lots of similar calls in apps and private apps where I couldn't immediately fix them.  The eventual fix is going to have to be a multi-step "add new APIs, then change software to use the new APIs, then hide and fix the data types" dance.